### DISCUSSION TITLE
Add-PnPGroupMember + Get-PnPGroupMember

### DISCUSSION THREAD

**Discussion by JaeEmAre67** [pnp/powershell#1130] [Q&A] [answered]
Is there a timing issue between adding a user to a group and getting that user from the group? I need to add a user to a group and then confirm that user was indeed added to the group. I've done

`Connect-PnPOnline -Url $whichSC;`
`Add-PnPGroupMember -group $addToGroup -EmailAddress $userToAdd;`
`Disconnect-PnPOnline;`
`Connect-PnPOnline -Url $whichSC;`
`$didAdd = Get-PnPGroupMember -Group $addToGroup -User $userToAdd;`
`$didAdd;`

which is always blank even though I can manually go to the site in question, open the group and see that the user was indeed added.

**Comment by veronicageek** [comment]
I don't think there is, I can run all at once with no problem. Unless you need it for some reason, the `Disconnect-PnPOnline` is also not necessary.

![image](https://user-images.githubusercontent.com/25181757/134934564-b0033bd1-5cda-48d3-92c1-16286581d8bd.png)

**Reply by KoenZomers** [reply]
There should indeed not be a delay. We're also not caching anything around these cmdlets. Does this consistently happen for you at multiple places?

**Comment by JaeEmAre67** [comment]
Thank you @veronicageek and @KoenZomers.

I only had the Disconnect-PnPOnline thinking that may be the issue. It's not there normally.

It happens on mutliple groups/site collections if that's what you mean.

**Comment by JaeEmAre67** [comment, marked as answer]
Our company's User info is different than just email due to some kind of ADFS connection. Putting in the entire info for User is returning as expected.

x:x#.x|xxxx|userEmail

Thank you
